### PR TITLE
fix(footer): fix styles when used on redhat.com

### PR DESCRIPTION
**Pull request by heyMP**
## What I did

1. Ported styles from `@patternfly/pfe-styles/pfe-base.css`.  `pfe-base.css` has a mixed bag of styles that have negative consequences.
2. This is a branch off of #302 

## Testing instructions

1. Start dev server `npm run dev`
3. Start the proxy server in a separate process `npm run proxy`
4. Open the the proxy url http://localhost:8001
5. Scroll to the bottom and you should see the rh-footer in context of the RHDC site.
6. Ensure it matches the footer in the dev server http://localhost:8000/demo/rh-footer/

## Todos

- [x] Port a styles from `pfe-base.css`
- [x] Add CSS overrides / resets for header & ul styles

### CODE REVIEW

**Comment by heyMP** [comment]
- [x] Port a styles from pfe-base.css
Screen Shot 2022-05-23 at 5 17 57 PM

File: elements/rh-footer/rh-footer-lightdom.css [hunk 1]
```diff
@@ -20,3 +20,21 @@ rh-footer :is([slot^=links], [slot=footer-links-primary], [slot=footer-links-sec
 rh-footer :is([slot^=links], [slot=footer-links-primary], [slot=footer-links-secondary]) a:is(:hover, :focus-within) {
   text-decoration: underline;
 }
+
+/* Via pfe-base.css */
+rh-footer a {
+  color: var(--pfe-broadcasted--link, #06c);
```
Comment (bennypowers) [inline elements/rh-footer/rh-footer-lightdom.css:26]: these should be `--rh-`, it's our party :)

```suggestion
  color: var(--rh-color-link-inline-on-light, var(--rh-color-blue-500, #06c));
```

File: elements/rh-footer/rh-footer-lightdom.css [hunk 2]
```diff
@@ -20,3 +20,21 @@ rh-footer :is([slot^=links], [slot=footer-links-primary], [slot=footer-links-sec
...
+  text-decoration: var(--pfe-broadcasted--link-decoration, none);
+}
+rh-footer a:hover {
+  color: var(--pfe-broadcasted--link--hover, #004080);
```
Comment (bennypowers) [inline elements/rh-footer/rh-footer-lightdom.css:30]:
```suggestion
  color: var(--rh-color-link-inline-hover-on-light, var(--rh-color-blue-600, #004080));
```

File: elements/rh-footer/rh-footer-lightdom.css [hunk 3]
```diff
@@ -20,3 +20,21 @@ rh-footer :is([slot^=links], [slot=footer-links-primary], [slot=footer-links-sec
...
+  text-decoration: var(--pfe-broadcasted--link-decoration--hover, underline);
+}
+rh-footer a:focus {
+  color: var(--pfe-broadcasted--link--focus, #004080);
```
Comment (bennypowers) [inline elements/rh-footer/rh-footer-lightdom.css:34]:
```suggestion
  color: var(--rh-color-link-inline-hover-on-light, var(--rh-color-blue-600, #004080));
```

File: elements/rh-footer/rh-footer-lightdom.css [hunk 4]
```diff
@@ -20,3 +20,21 @@ rh-footer :is([slot^=links], [slot=footer-links-primary], [slot=footer-links-sec
...
+  text-decoration: var(--pfe-broadcasted--link-decoration--focus, underline);
+}
+rh-footer a:visited {
+  color: var(--pfe-broadcasted--link--visited, #6753ac);
```
Comment (bennypowers) [inline elements/rh-footer/rh-footer-lightdom.css:38]:
```suggestion
  color: var(--rh-color-link-inline-visited-on-dark, var(--rh-color-purple-300, #a18fff));
```

File: elements/rh-footer/rh-footer-lightdom.css [hunk 5]
```diff
@@ -20,3 +20,21 @@ rh-footer :is([slot^=links], [slot=footer-links-primary], [slot=footer-links-sec
 rh-footer :is([slot^=links], [slot=footer-links-primary], [slot=footer-links-secondary]) a:is(:hover, :focus-within) {
   text-decoration: underline;
 }
+
+/* Via pfe-base.css */
+rh-footer a {
+  color: var(--pfe-broadcasted--link, #06c);
+  text-decoration: var(--pfe-broadcasted--link-decoration, none);
```
Comment (bennypowers) [inline elements/rh-footer/rh-footer-lightdom.css:27]:
```suggestion
  text-decoration: none;
```

File: elements/rh-footer/rh-footer-lightdom.css [hunk 6]
```diff
@@ -20,3 +20,21 @@ rh-footer :is([slot^=links], [slot=footer-links-primary], [slot=footer-links-sec
...
+}
+rh-footer a:hover {
+  color: var(--pfe-broadcasted--link--hover, #004080);
+  text-decoration: var(--pfe-broadcasted--link-decoration--hover, underline);
+}
+rh-footer a:focus {
+  color: var(--pfe-broadcasted--link--focus, #004080);
+  text-decoration: var(--pfe-broadcasted--link-decoration--focus, underline);
```
Comment (bennypowers) [inline elements/rh-footer/rh-footer-lightdom.css:35]:
```suggestion
  text-decoration: underline;
```

**Review by bennypowers — CHANGES_REQUESTED** [review]

**Review by bennypowers — APPROVED** [review]

**Pull request merged**